### ISSUE TITLE
Upgrade Trino to release 371

### ISSUE THREAD

**Issue by caldeirav** [os-climate/os_c_data_commons#116]
The latest Trino release contains a data access management feature we will likely require for our initial implementation of the Data Vault related functionalities:

https://github.com/trinodb/trino/pull/9991

We should upgrade with the latest version when possible and before we start building our prototype and demo for data vault (likely this coming sprint).

@MichaelTiemannOSC @erikerlandson @redmikhail @HeatherAck

**Comment by HumairAK** [comment]
@rimolive can we get an odh release image for trino for version 367?

**Comment by MichaelTiemannOSC** [comment]
FYI: https://trino.io/docs/current/release/release-368.html

Some comments about OOM fixes may relate to earlier problems I was seeing.

**Comment by MichaelTiemannOSC** [comment]
The upcoming Trino 369 has this interesting bugfix for Iceberg metadata: https://github.com/trinodb/trino/pull/10480

**Comment by MichaelTiemannOSC** [comment]
FYI: https://trino.io/docs/current/release/release-369.html from a few days ago.

**Comment by caldeirav** [comment]
Trino 371 image available at: https://quay.io/repository/opendatahub/trino?tab=tags&tag=371

**Comment by MichaelTiemannOSC** [comment]
This new handling of dropping tables/deletion was merged into Trino 373: https://github.com/trinodb/trino/pull/11062
